homeassistant/components/synology_dsm/binary_sensor.py

```diff
@@ -107,20 +107,36 @@ def __init__(
 class SynoDSMSecurityBinarySensor(SynoDSMBinarySensor):
     """Representation a Synology Security binary sensor."""
 
+    def __init__(
+        self,
+        api: SynoApi,
+        coordinator: SynologyDSMCentralUpdateCoordinator,
+    ) -> None:
+        """Initialize the Synology DSM security binary_sensor entity."""
+        description = SynologyDSMBinarySensorEntityDescription(
+            api_key=SynoCoreSecurity.API_KEY,
+            key="status",
+            translation_key="status",
+            device_class=BinarySensorDeviceClass.SAFETY,
+        )
+        super().__init__(api, coordinator, description)
+
     @property
     def is_on(self) -> bool:
         """Return the state."""
-        return getattr(self._api.security, self.entity_description.key) != "safe"  # type: ignore[no-any-return]
+        assert self._api.security is not None
+        return self._api.security.status != "safe"
 
     @property
     def available(self) -> bool:
         """Return True if entity is available."""
-        return bool(self._api.security)
+        return self._api.security is not None
 
     @property
     def extra_state_attributes(self) -> dict[str, str]:
         """Return security checks details."""
-        return self._api.security.status_by_check  # type: ignore[no-any-return]
+        assert self._api.security is not None
+        return self._api.security.status_by_check
```

Comment:
Not for now, but for a future PR, the dict should be defined in Home Assistant (and not passed blindly from the library)